### PR TITLE
fix: make yarn ios work again

### PR DESCRIPTION
**Pull request by MounirDhahri**
### Description
This PR makes the `yarn ios` command work again. This is useful if you hate opening xCode as much as I do. 

Screenshot 2022-11-09 at 10 13 39

### PR Checklist

- [x] I tested my changes on **iOS**
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- fix: make yarn ios work again - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by lordkiz — DISMISSED** [review]
Niceeee. I had thought `npx react-native run-ios` would start working with the new app shell.
Thanks for this.

**Comment by pvinis** [comment]
yea.. weird. what was it doing for you mounir before your changes?

**Comment by lordkiz** [comment]
> yea.. weird. what was it doing for you mounir before your changes?

@pvinis for me I would always have to open xcode and build from there. The yarn commands weren't working for me. I can't remember the exact reason why

**Comment by MounirDhahri** [comment]
`npx react-native run-ios` wasn't working for me before, let me check if it's back working. I didn't check it again 😄

**Comment by MounirDhahri** [comment]
It failed :/ let me see if I can tweak it

**Review by gkartalis — APPROVED** [review]
🔥❤️🔥❤️🤩 super useful! Thanks!

**Comment by pvinis** [comment]
ok so quick check, for me `yarn ios` complains that iPhone 13 sim is not there. then If i run `xcrun simctl list devices` i see i have iPhone 14 pro, and then this works `yarn ios --simulator="iPhone 14 Pro"`.

meanwhile, your changes are building, so ill see how that behaves once it finishes.

**Comment by pvinis** [comment]
ill offer a counter-suggestion, because i feel your changes are not going to help the rest of the devs 🤔. it builds for me, but it does not start. how about this:
we can change the `"ios"` script to something like this: `react-native run-ios --udid="$(xcrun simctl list | awk -F'[()]' '/(Booted)/ { print $2 }')"`?

the main bit is the `react-native run-ios` as before, but now we add a udid of the simulator we want it to run on. and to get the udid we use `xcrun simctl list` to get all the sims, and then with awk we get the sim that is booted (open and running right now), and we extract the udid.

this works perfectly for me. if that makes sense and it works for you all too, id go for that one.

try it now with just `npx react-native run-ios --udid="$(xcrun simctl list | awk -F'[()]' '/(Booted)/ { print $2 }')"`, and let me know, and we can move it to the script.

one note here, a simulator must be booted. so you can just start the Simulator.app and from the top menu just File>Open Simulator>pick any in there you like.

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "doctor": "./scripts/doctor.js",
     "generate-cities-cache": "node scripts/generate-cities-cache.js",
     "generate-cities-objc": "node scripts/generate-cities-objc.js",
-    "ios": "react-native run-ios",
+    "ios": "cd ios; xcodebuild build -workspace Artsy.xcworkspace -scheme Artsy; cd ..",
```
Comment (pvinis) [inline package.json:24]:
```suggestion
    "ios": "react-native run-ios --udid="$(xcrun simctl list | awk -F'[()]' '/(Booted)/ { print $2 }')",
```

just adding here the suggestion I offered, so its ready. please check if that is easier and accept or close 👍

Comment (pvinis) [reply package.json:24]: ew those `\"` lol. i dont like package.json scripts a lot 😅. if we like this, at some point I would just move it to a `./scripts/start-ios` or something.

Comment (lordkiz) [reply package.json:24]: @pvinis will this not necessitate that you have a simulator already running?

Comment (lordkiz) [reply package.json:24]: Sorry, Just saw your comment. Noted

Comment (pvinis) [reply package.json:24]: yep. we can add `yarn open-sim` in front of it.

**Comment by MounirDhahri** [comment]
nice, I like the new changes. Using `react-native run-ios` definitely makes more sense. Thanks for figuring it out @pvinis and everyone else for the review

**Review by pvinis — APPROVED** [review]

**Comment by pvinis** [comment]
tagging @patrinoua because i know she uses that command `yarn ios` a lot.

**Pull request merged**